helix-view/src/editor.rs

```diff
@@ -900,9 +900,14 @@ enum ThemeAction {
 }
 
 #[derive(Debug, Clone)]
-pub struct CompleteAction {
-    pub trigger_offset: usize,
-    pub changes: Vec<Change>,
+pub enum CompleteAction {
+    Applied {
+        trigger_offset: usize,
+        changes: Vec<Change>,
+    },
+    /// A savepoint of the currently active completion. The completion
+    /// MUST be restored before sending any event to the LSP
+    Selected { savepoint: Arc<SavePoint> },
```

Comment:
I am a bit confused with reading this name, what is selected? Only a bit clearer when I read the long comments above.

Reply:
`CompleteAction` represents the last action in the completion menu. You could either have:
* selected a completion (c-n, c-p, tab)
* or applied a completion (ret or any other key haver a completion was selected)
that's what this enum represents. If the completion menu is closed/replaced for whatever reason we must process this last action. For selected we must restore the savepoint and for Applied we must record the change so we can repeat it (existing code)